core/arch/x86/encode.c

```diff
@@ -844,6 +853,12 @@ reg_size_ok(decode_info_t *di /*prefixes field is IN/OUT; x86_mode is IN*/, reg_
         }
         return true;
     }
+    if (optype == TYPE_K_REG || optype == TYPE_K_MODRM || optype == TYPE_K_MODRM_R ||
+        optype == TYPE_K_VEX) {
+        return (opsize == OPSZ_1 || opsize == OPSZ_2 || opsize == OPSZ_4 ||
+                opsize == OPSZ_8);
+    }
+    /* XXX i#1312: TYPE_K_EVEX will be of variable size. */
```

Comment:
"will be": but I thought this PR was adding support for that type?  What makes the existing size_ok() check not work there?

Reply:
That's a comment for TYPE_K_EVEX. This type is not used. The instructions in this patch are just the VEX versions. The much larger set of EVEX instructions uses the mask as an actual mask, and the register is encoded in one of the EVEX bytes (evex.aaa). This is the type to be used. It's not 100% clear to me yet whether we need a variable type or if a fixed type will be ok. First I thought we need a variable type but now I think we will have 4 fixed size types again, because I think the size should always be defined by the opcode.